api/src/main/java/org/hawkular/inventory/api/model/Path.java

```diff
@@ -834,17 +834,23 @@ private boolean fillPossiblePathsToTarget(Class<?> targetType, Class<?> currentT
 
             if (targetType.equals(currentType)) {
                 if (isStart) {
-                    result.add(currentType);
+                    if (!result.contains(currentType)) {
```

Comment:
Is this necessary? `result` is a set, so a straight add should actually be cheaper than a check + add - both will try to look up if the element is present.